src/core/types/base/mempool.ts

```diff
@@ -98,17 +107,50 @@ export function decodeMempool(mempool: Mempool, ignoreAddresses: Record<string,
  * @param ignoreAddresses: Array containing wallet addresses that should be ignored and filtered.
  * @returns Boolean stating if `msg` is to be processed or not.
  */
-function isAllowedMempoolMsg(msg: MsgExecuteContract, ignoreAddresses: Record<string, boolean>): boolean {
+function isAllowedMempoolMsg(
+	msg: MsgExecuteContract,
+	ignoreAddresses: IgnoredAddresses,
+	timeoutDur: number,
+	iteration: number,
+): boolean {
 	//if the sender of the message is in our ignore list: skip this message
 	if (ignoreAddresses[msg.sender]) {
-		return false;
+		return !keepIgnored(msg.sender, undefined, ignoreAddresses, timeoutDur, iteration);
 		// if they use a contract to fund new wallets
 	} else if (ignoreAddresses[msg.contract]) {
 		const containedMsg = JSON.parse(fromUtf8(msg.msg));
-		const gets = fromAscii(fromBase64(containedMsg.delegate.msg));
-		ignoreAddresses[gets] = true;
+		if (containedMsg.delegate) {
+			const gets = fromAscii(fromBase64(containedMsg.delegate.msg));
+			return !keepIgnored(msg.sender, gets, ignoreAddresses, timeoutDur, iteration);
+		}
+	}
+	return true;
+}
+/**
+ *
+ */
+function keepIgnored(
+	address: string,
+	reciever: string | undefined,
+	ignoreAddresses: IgnoredAddresses,
+	timeoutDur: number,
+	iteration: number,
+) {
+	if (
+		ignoreAddresses[address].source ||
```

Comment:
is this ever false? if it's false, why is the address still in the list?

Reply:
i think its possible atm that an address sends a tx and gets timeouted because the arb opportunity failed. timeoutAt = 1 timeoutDur = 100. If the address isnt active for the next 100 blocks and clearIgnoreAddresses isnt called after that for like 150 blocks more it stays saved as object, even though it shouldnt be timeouted anymore. The address could send an arbitrage opportunity now but we would still ignore it. So if an ignored address sends a tx  we check if the address should still be timeouted. 
So thats what it should do. If you see an error let me know!